src/index/base.cpp

```diff
@@ -206,22 +215,19 @@ void BaseIndex::ThreadSync()
 
 bool BaseIndex::Commit()
 {
-    CDBBatch batch(GetDB());
-    if (!CommitInternal(batch) || !GetDB().WriteBatch(batch)) {
-        return error("%s: Failed to commit latest %s state", __func__, GetName());
-    }
-    return true;
-}
-
-bool BaseIndex::CommitInternal(CDBBatch& batch)
-{
-    LOCK(cs_main);
     // Don't commit anything if we haven't indexed any block yet
     // (this could happen if init is interrupted).
     if (m_best_block_index == nullptr) {
         return false;
```

Comment:
ddbbdcf4112c3a84e3287e6bf701e0eb592b24c3: This changes behavior: Previously there was a call to `error()`, now there is none?